### PR TITLE
Alcor Database Access module document

### PR DESCRIPTION
**Pull request by pkommoju**
* Document describing the DB module in Alcor. Generic facilities and Ignite specific internals are documented.

### CODE REVIEW

File: docs/modules/ROOT/pages/db_services/alcor-database-interface.adoc [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+.. SQL Schema name
+.. QueryEntity
+.. QueryIndex
+
+. SqlFieldsQuery
+. QueryCursor
+. ScanCursor
+. IgniteBiPredicate
+. Transaction
+
```
Comment (xieus) [inline docs/modules/ROOT/pages/db_services/alcor-database-interface.adoc:146]: As discussed, we are adding a Usage section to show code snippet for major use cases.

**Review by xieus — COMMENTED** [review]
@pkommoju Here is the first batch of comments.

File: docs/modules/ROOT/pages/db_services/alcor-database-interface.adoc [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+==== ICahce
+
+An abstraction of a cache entity. It is parameterized by the type of the key and the type of the value stored in the cache, in other words the, an entry in the cache. It provides the following interfaces to work the the cache.
+
+.. put - Given a key and value, store them in the cache, replace the value if the key already exists.
+.. putAll - Given a set of keys and corresponding values, store them in the cache in a single operation.
+.. putIfAbsent - Like put but only if the key does not exist.
+.. containsKey - Given a key, return true if it exists in the cache, false otherwise.
+.. get - Given a key, return the value if a match is found.
+.. get - Given a set of keys return values matching those keys.
```
Comment (DavidLiu506) [inline docs/modules/ROOT/pages/db_services/alcor-database-interface.adoc:26]: getAll()

File: docs/modules/ROOT/pages/db_services/alcor-database-interface.adoc [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+=== Checking if an entry exists: containsKey()
+
+If just checking if an exists or not is sufficient, containsKey() method is used. This is avoids reading the value and discarding it.
+
+=== Querying: get, getAll with or without any qualifying conditions.
+
+. get(key): Return the entry matching the given key. This is also called point lookup.
+. get(set of keys): Return all entries which have keys matching keys. This is a bulk query.
+. get(qualifying conditions): This is used when cache lookup is based on non key fields. DB module can decide, with the help of the information collected during the creation of the cache, if a SQL Index query can be used to speed up the query execution or not.
```
Comment (DavidLiu506) [inline docs/modules/ROOT/pages/db_services/alcor-database-interface.adoc:105]: getAll()

**Review by xieus — APPROVED** [review]

**Pull request merged**